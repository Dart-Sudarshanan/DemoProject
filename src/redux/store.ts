import { configureStore } from "@reduxjs/toolkit";
import authSlice from "./slices/authSlice";

import { reducer as formReducer } from "redux-form";
import userSlice from "./slices/userSlice";
import counterSlice from "./slices/counterSlice";

const store = configureStore({
  reducer: {
    form: formReducer,
    userAuth: authSlice.reducer,
    userList: userSlice,
    counterList: counterSlice,
  },
});

export type ApplicationState = ReturnType<typeof store.getState>;

export const authUser = (state: ApplicationState) => state.userAuth;
export const userList = (state: ApplicationState) => state.userList;
export const counterList = (state: ApplicationState) => state.counterList;

export { store };
