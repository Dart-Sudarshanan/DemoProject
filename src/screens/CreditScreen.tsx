import { NativeStackScreenProps } from "@react-navigation/native-stack";
import React from "react";

import { Pressable, ScrollView, StyleSheet, Text, View } from "react-native";
import { DrawerStackParams } from "../components/navigation_component/PrivateScreens";

type Props = NativeStackScreenProps<DrawerStackParams, "Credit">;

const CreditScreen = (props: Props) => {
  return (
    <ScrollView>
      <View style={styles.sectionContainer}>
        <View style={styles.contentWrapper}>
          <Text style={styles.text}>Credits</Text>
          <Pressable style={styles.button} onPress={() => props.navigation.navigate("About")}>
            <Text style={styles.buttonTxt}>About</Text>
          </Pressable>
          <Pressable style={[styles.button, { marginTop: 5 }]} onPress={() => props.navigation.navigate("Home")}>
            <Text style={styles.buttonTxt}>Home</Text>
          </Pressable>
        </View>
      </View>
    </ScrollView>
  );
};
const styles = StyleSheet.create({
  sectionContainer: {
    marginTop: 10,
    paddingHorizontal: 15,
  },
  contentWrapper: {
    alignItems: "center",
    justifyContent: "center",
  },
  text: {
    fontSize: 24,
    fontWeight: "400",
  },
  button: {
    paddingHorizontal: 40,
    paddingVertical: 10,
    borderRadius: 4,
    backgroundColor: "#4c51e6",
  },
  buttonTxt: {
    fontSize: 18,
    fontWeight: "500",
    color: "#fff",
  },
});

export default CreditScreen;
