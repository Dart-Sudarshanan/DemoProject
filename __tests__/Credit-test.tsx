import React from "react";
import { Provider } from "react-redux";

import renderer from "react-test-renderer";
import { store } from "../src/redux/store";
import CreditScreen from "../src/screens/CreditScreen";

it("Credit screen renders correctly", () => {
  const navigation = { navigate: () => {} };
  spyOn(navigation, "navigate");
  const tree = renderer.create(
    <Provider store={store}>
      <CreditScreen />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
