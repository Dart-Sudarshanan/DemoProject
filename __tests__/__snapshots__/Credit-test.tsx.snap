// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit screen renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "marginTop": 10,
          "paddingHorizontal": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "400",
            }
          }
        >
          Credits
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#4c51e6",
              "borderRadius": 4,
              "paddingHorizontal": 40,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            About
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#4c51e6",
                "borderRadius": 4,
                "paddingHorizontal": 40,
                "paddingVertical": 10,
              },
              Object {
                "marginTop": 5,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            Home
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
